app/baseapp.go

```diff
@@ -231,6 +239,29 @@ func (app *BaseApp) setDeliverState(header abci.Header) {
 	}
 }
 
+// setConsensusParams memoizes the consensus params.
+func (app *BaseApp) setConsensusParams(consensusParams *abci.ConsensusParams) {
+	app.consensusParams = consensusParams
+}
+
+// setConsensusParams stores the consensus params to the main store.
+func (app *BaseApp) storeConsensusParams(consensusParams *abci.ConsensusParams) {
+	consensusParamsBz, err := proto.Marshal(consensusParams)
+	if err != nil {
+		panic(err)
+	}
+	mainStore := app.cms.GetKVStore(protocol.KeyMain)
```

Comment:
What's the difference between `protocol.KeyMain` and  `mainKey` in the above code:
 `mainStore := app.cms.GetKVStore(mainKey)`